src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,141 @@
+     * If the data file does not exist, a new one will be created and
+     * an empty ArrayList will be returned.
+     *
+     * @return An ArrayList of Tasks with the initial tasks added in where applicable.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            File directory = new File("./data");
+            if (!directory.exists()) {
+                directory.mkdir();
+            }
+            File inputFile = new File("./data/tasks.txt");
+            if (inputFile.createNewFile()) {
+                // new file created
+            } else {
+                list = readTasks(inputFile);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        } finally {
+            return list;
+        }
+    }
+
+    /**
+     * Read task from the filePath.
+     *
+     * @param inputFile File which the tasks are read from.
+     * @return An ArrayList of Tasks which the tasks read are added into.
+     */
+    private ArrayList<Task> readTasks(File inputFile) {
+        ArrayList<Task> list = new ArrayList<>();
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
+        try {
+            Scanner s = new Scanner(inputFile);
+            BufferedReader reader = new BufferedReader(new FileReader(inputFile));
+            while (s.hasNextLine()) {
+                String line = s.nextLine();
+                String type = line.substring(1, 2);
+                String mark = line.substring(4, 5);
+
+                if (type.equals("T")) {
+                    String description = line.substring(7);
+                    Todo t = new Todo(description);
+                    if (mark.equals("X")) {
+                        t.markAsDone();
+                    }
+                    list.add(t);
+                } else if (type.equals("D")) {
+                    String temp = line.substring(7);
+                    String[] result = temp.split("\\(by: ");
+                    String description = result[0].trim();
+                    String by = result[1].substring(0, result[1].length() - 1);
+                    LocalDate date = LocalDate.parse(by.substring(0, 11), dateTimeFormatter);
+                    LocalTime time = LocalTime.parse(by.substring(12, 17));
+                    Deadline t = new Deadline(description, date, time);
+                    if (mark.equals("X")) {
+                        t.markAsDone();
+                    }
+                    list.add(t);
+                } else if (type.equals("E")) {
+                    String temp = line.substring(7);
+                    String[] temp2 = temp.split("\\(from: ");
+                    String description = temp2[0];
+                    String result = temp2[1];
+                    LocalDate startDate = LocalDate.parse(result.substring(0, 11), dateTimeFormatter);
```

Comment:
perhaps can use the LocalDateTime class to store the start and end of an event instead since you want to keep track of both date and time